File: backend/src/main/java/bizworks/backend/dtos/LeaveType.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package bizworks.backend.dtos;

/**
 *
 * @author PC
 */
public enum LeaveType {
    SICK("Sick leave"),
    MATERNITY("Maternity leave"),
    PERSONAL("Personal leave"),
    BEREAVEMENT("Bereavement leave"),
    MARRIAGE("Marriage leave"),
    CIVIC_DUTY("Civic duty leave"),
    OTHER("Other leave");

    private final String description;

    LeaveType(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }
}

File: mobile/mobile/lib/models/LeaveType.dart
enum LeaveType {
  SICK,
  MATERNITY,
  PERSONAL,
  BEREAVEMENT,
  MARRIAGE,
  CIVIC_DUTY,
  OTHER,
}

extension LeaveTypeExtension on LeaveType {
  String get description {
    switch (this) {
      case LeaveType.SICK:
        return "Sick leave";
      case LeaveType.MATERNITY:
        return "Maternity leave";
      case LeaveType.PERSONAL:
        return "Personal leave";
      case LeaveType.BEREAVEMENT:
        return "Bereavement leave";
      case LeaveType.MARRIAGE:
        return "Marriage leave";
      case LeaveType.CIVIC_DUTY:
        return "Civic duty leave";
      case LeaveType.OTHER:
        return "Other leave";
      default:
        return "";
    }
  }
}
